### PR TITLE
Formalizing the typings

### PR DESCRIPTION
**Pull request by engineersamuel**

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {ICase_Comment__c, ICase_Comment__c_fields} from "./models/case";
+
+export = hydrajs;
+export as namespace hydrajs;
+
+declare module hydrajs {
```
Comment (jtrantin) [inline src/index.d.ts]: @engineersamuel Ah, so in the end you do import the interfaces in the module def. Are those available without importing then? Or do we need to import ICase_Comment__c manually in Ascension?

Comment (engineersamuel) [inline src/index.d.ts]: @jtrantin  http://cloud.samuelmendenhall.com/2B1U1534161n :)  We will have to manually import the `__fields` to get that type checking, but that is expected.

Comment (jtrantin) [inline src/index.d.ts]: Yeah, in that case it's obvious, but what if I want to create a new object of type `IComment_Case__c` (e.g. when faking a change in reducer, or when creating object to be sent to the API), then I have to import the interface from /src/..., right?

Comment (engineersamuel) [inline src/index.d.ts]: That's right, you'd have to import the `src/models/case.ts`  I did try to embed the interface in the index.d.ts to see if that would resolve it but it won't.  I think that makes sense considering that the d.ts defines the typings for the library, but if you specify `IComment_Case__c` then that is treated as an independent variable within the context of the file itself and the compiler doesn't know what that means, it is just a free standing variable without some variable to define it, so you'd have to import it.  There could be some incantation with the index.d.ts that could give it implicitly, not sure, feel free to play with it in `ascension/node_modules/hydrajs/src/index.d.ts`

Comment (jtrantin) [inline src/index.d.ts]: I agree with your assessment. Having it auto-imported could potentially cause conflicts.

**Pull request merged**